pkg/e2e/verify/check_workloads.go

```diff
@@ -0,0 +1,133 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var onNodesTestName string = "[Suite: informing] [OSD] pod validating webhook"
+
+func init() {
+	alert.RegisterGinkgoAlert(onNodesTestName, "SD-SREP", "Matt Bargenquast", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(onNodesTestName, func() {
+
+	h := helper.New()
+
+	ginkgo.Context("worker nodes", func() {
+
+		ginkgo.It("on worker nodes", func() {
+
+			_, infra, err := checkWorkloadsOnWorkerNodes(h.Kube().CoreV1())
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = checkPods(h.Kube().CoreV1(), infra)
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+	})
+})
+
+func checkPods(podClient v1.CoreV1Interface, infra []string) (map[string]string, error) {
+
+	violators := make(map[string]string)
+
+	operators := map[string]string{"cloud-ingress-operator": "openshift-cloud-ingress-operator",
+		"configure-alertmanager-operator": "openshift-monitoring",
+		"custom-domains-operator":         "openshift-custom-domains-operator",
+		"managed-upgrade-operator":        "openshift-managed-upgrade-operator",
+		"managed-velero-operator":         "openshift-velero",
+		"must-gather-operator":            "openshift-must-gather-operator",
+		"osd-metrics-exporter":            "openshift-osd-metrics",
+		"rbac-permissions-operator":       "openshift-rbac-permissions"}
+
+	listOpts := metav1.ListOptions{}
+	list, err := podClient.Pods(metav1.NamespaceAll).List(context.TODO(), listOpts)
+	if err != nil {
+		return nil, fmt.Errorf("error getting pod list: %v", err)
+	}
+
+	if len(list.Items) == 0 {
+		return nil, fmt.Errorf("pod list is empty. this should NOT happen")
+	}
+
+	for _, pod := range list.Items {
+		//For each pod.ObjectMeta.Name, pod.Spec.NodeName we will check if they are a match for the operaor namespace and pod name prefix.
+		for op, namespace := range operators {
+			if pod.ObjectMeta.Namespace == namespace {
+				if strings.HasPrefix(pod.ObjectMeta.Name, op) {
+
+					if !stringInSlice(pod.Spec.NodeName, infra) {
+
+						violators[pod.ObjectMeta.Name] = pod.Spec.NodeName
+						log.Printf(" Violation detected: pod %s, doesn't run on infra node but instead runs on %s", pod.ObjectMeta.Name, pod.Spec.NodeName)
+					}
+				}
+
+			}
+
+		}
+
+	}
+
+	if len(violators) > 0 {
+		return violators, fmt.Errorf("Found infrastructure pods that do not run on infra nodes %v.", err)
+	}
+
+	//log.Printf("Pods %v", filteredPods)
+
+	return violators, nil
+}
+
+func checkWorkloadsOnWorkerNodes(nodeClient v1.CoreV1Interface) ([]string, []string, error) {
```

Comment:
```suggestion
func getWorkerNodesByType(nodeClient v1.CoreV1Interface) ([]string, []string, error) {
```
Can we call this something else? It doesn't "check" the workloads so much as filter the list of nodes to those with a specific role.

Reply:
I'll rename it, thanks for the suggestion.